platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/InterruptibleCompletableFuture.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.merkledb;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+/**
+ * A wrapper around {@link CompletableFuture} that supports interruption
+ * when the CompletableFuture is canceled.
+ *
+ */
+public class InterruptibleCompletableFuture<T> {
```

Comment:
The main idea behind this class is to overcome `CompletableFuture` shortcoming  - it doesn't support cancellation with a thread interruption. And compaction is implemented so that we actually need the interruption mechanism to stop it as soon as possible. So, I decided to come with the workaround.

Reply:
Thread interrupt generally should exit the entire process (it's quite rare it works differently, as source of interrupt isn't known, so anything that prevents that can prevent exiting the process cleanly, and cause it to require process termination to stop).
I see what you're wanting, but I suspect we should be telling the compaction task to exit, rather than trying to terminate the thread it's running on.
A short discussion might work to come up with a cleaner approach that avoids interrupting threads (and potentially creating a larger impact than intended).